### PR TITLE
Feat/3836 api defint camdecmps schema entity relationships

### PR DESCRIPTION
**Pull request by fanwari1**
## Pull Request

This PR adds entity relationships.

The primary key fields are updated from the name of the field to being "id" since this is what's more consistent with the rest of our projects. 

The code fields are also updated from ending in "Cd" to "Code" to match our projects.

### CODE REVIEW

File: src/entities/nsps4t-annual.entity.ts [hunk 1]
```diff
@@ -44,4 +47,27 @@ export class Nsps4tAnnual extends BaseEntity {
...
+  location: MonitorLocation;
+
+  @ManyToOne(
+    () => ReportingPeriod,
+    o => o.nsps4tAnnuals,
+  )
+  @JoinColumn({ name: 'rpt_period_id' })
+  reportingPeriod: ReportingPeriod;
+  
+  // @TODO: Verify this relationship type once there is data in nsps4t-annual and nsps4t-summary tables
```
Comment (fanwari1) [inline src/entities/nsps4t-annual.entity.ts:65]: There is no data for these nsps4t tables so I could not verify that the relationship is actually one to one. I decided to leave this here for the field `nsps4tSummary` and update if it's needed later on or I can just remove this and add back once the data comes in and we can verify the relationship. There are three others below as well.

File: src/entities/component.entity.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  @Column({ type: 'varchar', length: 45, nullable: false, name: 'mon_loc_id' })
+  monLocId: string;
+
+  @Column({
+    type: 'varchar',
+    length: 3,
+    nullable: false,
+    name: 'component_identifier',
+  })
+  componentID: string;
```
Comment (eglove) [inline src/entities/component.entity.ts:31]: Should this be componentId to follow camel casing?

File: src/entities/daily-test-summary.entity.ts [hunk 3]
```diff
@@ -55,8 +60,50 @@ export class DailyTestSummary extends BaseEntity {
...
+    o => o.dailyTestSummaries,
+  )
+  @JoinColumn({ name: 'rpt_period_id' })
+  reportingPeriod: ReportingPeriod;
+  
+  @ManyToOne(
+    () => Component,
+    o => o.dailyTestSummaries,
+  )
+  @JoinColumn({ name: 'component_id' })  @OneToOne(
```
Comment (eglove) [inline src/entities/daily-test-summary.entity.ts]: Seems like an issue here, component: Component doesn't have a relationship decorator and dailyTestSummary: DailTestSummary has two.

File: src/entities/monitor-formula.entity.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+  @Column({ type: 'varchar', length: 7, nullable: false, name: 'parameter_cd' })
+  parameterCode: string;
+
+  @Column({ type: 'varchar', length: 7, nullable: true, name: 'equation_cd' })
+  formulaCode: string;
+
+  @Column({
+    type: 'varchar',
+    length: 3,
+    nullable: true,
```
Comment (eglove) [inline src/entities/monitor-formula.entity.ts]: It looks like this should not be nullable.

File: src/entities/monitor-formula.entity.ts [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    transformer: new NumericColumnTransformer(),
+  })
+  endHour: number;
+
+  @Column({
+    type: 'varchar',
+    length: 200,
+    name: 'formula_equation',
+  })
+  formulaText: string;
```
Comment (eglove) [inline src/entities/monitor-formula.entity.ts:62]: Default nullable is false, this should be nullable: true

File: src/entities/monitor-formula.entity.ts [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+  @Column({ type: 'varchar', length: 8, nullable: false, name: 'userid' })
```
Comment (eglove) [inline src/entities/monitor-formula.entity.ts]: This looks like it should be nullable: true and varchar(25)

File: src/entities/monitor-location.entity.ts [hunk 7]
```diff
@@ -0,0 +1,161 @@
...
+export class MonitorLocation extends BaseEntity {
+  @PrimaryColumn({
+    name: 'mon_loc_id',
+  })
+  id: string;
+
+  @Column({
+    name: 'unit_id',
+    nullable: true,
+  })
```
Comment (eglove) [inline src/entities/monitor-location.entity.ts:35]: unit_id is numeric(38)

File: src/entities/monitor-location.entity.ts [hunk 8]
```diff
@@ -0,0 +1,161 @@
...
+
+  @Column({
+    name: 'userid',
+    nullable: true,
+    type: 'date',
+  })
+  addDate: Date;
+
+  @Column({
+    name: 'userid',
```
Comment (eglove) [inline src/entities/monitor-location.entity.ts:58]: name: 'userid' looks like it should be 'update_date'

File: src/entities/monitor-system.entity.ts [hunk 9]
```diff
@@ -0,0 +1,127 @@
...
+import { HrlyParamFuelFlow } from './hrly-param-fuel-flow.entity';
+
+@Entity({ name: 'camdecmps.monitor_system' })
+export class MonitorSystem extends BaseEntity {
+  @PrimaryColumn({ type: 'varchar', length: 45, name: 'mon_sys_id' })
+  id: string;
+
+  @Column({ type: 'varchar', length: 45, nullable: false, name: 'mon_loc_id' })
+  locationId: string;
+  nsps4tCompliancePeriods
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:20]: Should this be here?

File: src/entities/monitor-system.entity.ts [hunk 10]
```diff
@@ -0,0 +1,127 @@
...
+  @Column({ type: 'varchar', length: 7, nullable: true, name: 'sys_type_cd' })
+  systemTypeCode: string;
+
+  @Column({
+    type: 'varchar',
+    length: 3,
+    nullable: true,
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:27]: system_identifier should be nullable: false

File: src/entities/monitor-system.entity.ts [hunk 11]
```diff
@@ -0,0 +1,127 @@
...
+
+@Entity({ name: 'camdecmps.monitor_system' })
+export class MonitorSystem extends BaseEntity {
+  @PrimaryColumn({ type: 'varchar', length: 45, name: 'mon_sys_id' })
+  id: string;
+
+  @Column({ type: 'varchar', length: 45, nullable: false, name: 'mon_loc_id' })
+  locationId: string;
+  nsps4tCompliancePeriods
+  @Column({ type: 'varchar', length: 7, nullable: true, name: 'sys_type_cd' })
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:21]: system_type_cd is nullable: false

File: src/entities/monitor-system.entity.ts [hunk 12]
```diff
@@ -0,0 +1,127 @@
...
+    length: 3,
+    nullable: true,
+    name: 'system_identifier',
+  })
+  monitoringSystemId: string;
+
+  @Column({
+    type: 'varchar',
+    length: 7,
+    nullable: false,
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:35]: nullable: true

File: src/entities/monitor-system.entity.ts [hunk 13]
```diff
@@ -0,0 +1,127 @@
...
+    name: 'sys_designation_cd',
+  })
+  systemDesignationCode: string;
+
+  @Column({ type: 'varchar', length: 7, nullable: false, name: 'fuel_cd' })
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:40]: nullable: true

File: src/entities/monitor-system.entity.ts [hunk 14]
```diff
@@ -0,0 +1,127 @@
...
+  fuelCode: string;
+
+  @Column({ type: 'date', nullable: false, name: 'begin_date' })
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:43]: nullable: true

File: src/entities/monitor-system.entity.ts [hunk 15]
```diff
@@ -0,0 +1,127 @@
...
+  beginDate: Date;
+
+  @Column({ type: 'date', nullable: true, name: 'end_date' })
+  endDate: Date;
+
+  @Column({ name: 'begin_hour', transformer: new NumericColumnTransformer() })
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts:49]: Default is nullable: false, this should be nullable: true

File: src/entities/monitor-system.entity.ts [hunk 16]
```diff
@@ -0,0 +1,127 @@
...
+  beginHour: number;
+
+  @Column({ name: 'end_hour', transformer: new NumericColumnTransformer() })
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts]: Default is nullable: false, this should be nullable: true

File: src/entities/monitor-system.entity.ts [hunk 17]
```diff
@@ -0,0 +1,127 @@
...
+  endHour: number;
+
+  @Column({ type: 'varchar', nullable: true, length: 8, name: 'userid' })
```
Comment (eglove) [inline src/entities/monitor-system.entity.ts]: Looks like this should be varchar(25)

File: src/entities/reporting-period.entity.ts [hunk 18]
```diff
@@ -0,0 +1,178 @@
...
+    name: 'period_abbreviation',
+    nullable: false,
+  })
+  periodAbbreviation: string;
+
+  @Column({
+    name: 'archive_ind',
+    nullable: false,
+  })
+  archiveInd: number;
```
Comment (eglove) [inline src/entities/reporting-period.entity.ts:79]: needs NumericColumnTransformer

File: src/entities/weekly-system-integrity.entity.ts [hunk 19]
```diff
@@ -1,16 +1,17 @@
-import { BaseEntity, Column, Entity, PrimaryColumn } from 'typeorm';
+import { BaseEntity, Column, Entity, JoinColumn, OneToOne, PrimaryColumn } from 'typeorm';
 import { NumericColumnTransformer } from '@us-epa-camd/easey-common/transforms';
+import { WeeklyTestSummary } from './weekly-test-summary.entity';
 
 @Entity({ name: 'weekly_system_integrity' })
```
Comment (eglove) [inline src/entities/weekly-system-integrity.entity.ts:5]: Just noticed this when trying to find the table, does this need to be campdecmps.weekly_system_integrity?

File: src/entities/weekly-system-integrity.entity.ts [hunk 20]
```diff
@@ -60,7 +61,7 @@ export class WeeklySystemIntegrity extends BaseEntity {
   @Column({ name: 'add_date', nullable: true })
   addDate: Date;
 
-  @Column({ name: 'update_date', nullable: true })
+  @Column({ name: 'updatesamplingTrains_date', nullable: true })
```
Comment (eglove) [inline src/entities/weekly-system-integrity.entity.ts]: Looking at the db, this column name is update_date, was this meant for the typeorm display name?

**Review by eglove — CHANGES_REQUESTED** [review]

File: src/entities/component.entity.ts (hunk 2 above)
Comment (fanwari1) [reply src/entities/component.entity.ts:31]: I had it as ID instead of Id to keep consistency because in the other repos it's ID.

Comment (eglove) [reply src/entities/component.entity.ts:31]: wfm, just wanted to sure it's not a mistake

File: src/entities/daily-test-summary.entity.ts (hunk 3 above)
Comment (fanwari1) [reply src/entities/daily-test-summary.entity.ts]: Fixed. Good catch.

File: src/entities/weekly-system-integrity.entity.ts (hunk 19 above)
Comment (fanwari1) [reply src/entities/weekly-system-integrity.entity.ts:5]: Yes my mistake. Should be fixed now.

File: src/entities/weekly-system-integrity.entity.ts (hunk 20 above)
Comment (fanwari1) [reply src/entities/weekly-system-integrity.entity.ts]: I think I accidentally clicked in that area and pasted something. It should just be update_date. Should be fixed now.

**Review by eglove — APPROVED** [review]

**Pull request merged**